pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagentplan.go

```diff
@@ -17,10 +24,27 @@ func (h *handler) OnChangeInstallSUC(cluster *rancherv1.Cluster, status rancherv
 		return nil, status, nil
 	}
 
+	currentVersion, err := semver.NewVersion(cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return nil, status, err
+	}
+
+	// indicate to the SUC chart if we want to
+	// install PodSecurityPolicy manifests
+	pspEnabled := false
+	if currentVersion.LessThan(Kubernetes125) {
+		pspEnabled = true
+	}
+
+	// we must limit the output of name.SafeConcatName to at most 48 characters because
+	// a) the chart release name cannot exceed 53 characters, and
+	// b) upon creation of this resource the prefix 'mcc-' will be added to the release name, hence the limiting to 48 characters
+	managedChartName := name.Limit(name.SafeConcatName(cluster.Name, "managed", "system-upgrade-controller"), 48)
```

Comment:
what happened before when we had a super long cluster name...? is this going to break upgrades for super long cluster names?

Reply:
The bundle was never properly deployed, this shouldn't break any existing clusters and should actually result in the proper deployment of the bundle